### PR TITLE
anonymization: improve email sending

### LINKED ISSUE #731: Anonymization: decrease n sent emails
When the anonymization cron runs, it tried to "delete" users when they have no loans/orders. If they have, it will send an email to the librarians to take actions.

The system should not send one email per user, but only one summary email at the end.

### PR DESCRIPTION
**Pull request by ntarocco**
* send only one summary email instead of one per user when the cron
  cannot anonymize user because they have loans/orders
* closes #731

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@ function cleanup() {
 trap cleanup EXIT
 python -m check_manifest --ignore ".*-requirements.txt"
 python -m sphinx.cmd.build -qnNW docs docs/_build/html
+export POSTGRESQL_VERSION=12.4
```
Comment (ntarocco) [inline run-tests.sh:22]: What if we remove this?

File: tests/ldap/test_ldap.py [hunk 2]
```diff
@@ -547,5 +522,65 @@ def _prepare():
...
+    mocker.patch("invenio_app_ils.patrons.anonymization.current_app_ils.patron_indexer.delete")  # noqa
+    # mock RemoteAccounts
+    mock1 = Mock()
+    mock1.user_id = 1  # patron 1
+    mock2 = Mock()
+    mock2.user_id = 2  # patron 2
+    mocker.patch("cds_ils.ldap.api.remap_invenio_users",
+                 return_value=[mock1, mock2])
+    # mock that the any `InvenioUser` exist
+    mocker.patch("cds_ils.ldap.api.InvenioUser")
```
Comment (ntarocco) [inline tests/ldap/test_ldap.py:558]: is this working in the end?

**Comment by ntarocco** [comment]
LGTM, thanks a lot!

File: tests/ldap/test_ldap.py (hunk 2 above)
Comment (kpsherva) [reply tests/ldap/test_ldap.py:558]: yes I changed the mock from where it was implemented to where it was used, the same with the mock below

**Pull request merged**